ddp-workspace/projects/ddp-pancan/src/app/components/footer/footer.component.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { mockComponent, WindowRef } from 'ddp-sdk';
+import { RouterTestingModule } from '@angular/router/testing';
+import { of, Observable } from 'rxjs';
+import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
+import { By } from '@angular/platform-browser';
+import { FooterComponent } from './footer.component';
+import { MatIconModule } from '@angular/material/icon';
+
+class TranslateLoaderMock implements TranslateLoader {
+    getTranslation(code: string = ''): Observable<object> {
+        const TRANSLATIONS = {
+            en: {}
+        };
+        return of(TRANSLATIONS[code]);
+    }
+}
+
+describe('FooterComponent', () => {
+    let fixture: ComponentFixture<FooterComponent>;
+    let component: FooterComponent;
+    let nativeWindowSpy: jasmine.SpyObj<Window>;
+
+    beforeEach(async () => {
+        const auth = mockComponent({ selector: 'app-auth' });
+        const navigation = mockComponent({ selector: 'app-navigation' });
+        nativeWindowSpy = jasmine.createSpyObj('sessionSpy', ['scrollTo']);
+        await TestBed.configureTestingModule({
+            imports: [
+                RouterTestingModule,
+                TranslateModule.forRoot({ loader: { provide: TranslateLoader, useClass: TranslateLoaderMock }, }),
+                MatIconModule,
+                NoopAnimationsModule
+            ],
+            providers: [
+                { provide: WindowRef, useValue: { nativeWindow: nativeWindowSpy }},
+                { provide: 'toolkit.toolkitConfig', useValue: {} },
+            ],
+            declarations: [FooterComponent, auth, navigation],
+        })
+            .compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(FooterComponent);
+        component = fixture.debugElement.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create component', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should scroll to the top when click on back-to-top button', () => {
+        const backToTopButton = fixture.debugElement.query(By.css('.back-to-top')).nativeElement;
+        backToTopButton.click();
+        expect(nativeWindowSpy.scrollTo).toHaveBeenCalled();
```

Comment:
#minor

would be nice to check if it was a scroll exactly to the top.

Reply:
I tried, but typescript compiler kept throwing the error that the method should have 2 arguments instead of 1, though officially Window interface has 2 different types for scrollTo. I couldn't find the solution how to fix it. If you know - please let me know

Reply:
for example: 
> expect(nativeWindowSpy.scrollTo).toHaveBeenCalled();
        expect(nativeWindowSpy.scrollTo.calls.first().args[0]).toEqual({ top: 0, behavior: 'smooth'} as any);

Reply:
nice idea! I'll update the test in the next PR